### PR TITLE
Conditionally reducing dashboard chatter for users

### PR DESCRIPTION
**Pull request by jeremyf**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description

This commit provides two things:

1.  Some notes related to my analysis regarding the dashboard
2.  Conditional redirects and rendering based on article policies

The code comments say most of what I want to say, but to reiterate:

When a user can't create articles nor do they already have published
articles, then we don't want to avoid showing them stats related to
articles.

## Related Tickets & Documents

Closes forem/forem#16913
Related to forem/forem#16908 and forem/forem#16931

## QA Instructions, Screenshots, Recordings

If you want to reproduce this in the browser, you'll need to:

`rails runner "FeatureFlag.enable(::limit_post_creation_to_admins)"`

Then login as a user who has no articles, go to `/dashboard` and you should be
redirected to `/dashboard/following_tags`.  You will also not see Posts,
Series, nor Followers on the side panel.

Now, run `rails runner "FeatureFlag.disable(::limit_post_creation_to_admins)"`.
And refresh your user's page.  You should see Posts, Series, and Followers.
Create and publish an article.

Then run `rails runner "FeatureFlag.enable(::limit_post_creation_to_admins)"`
and go to `/dashboard`.  You should remain on the "Dashboard" and see Posts,
Series, and Followers.

### UI accessibility concerns?

None.

## Added/updated tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams

### CODE REVIEW

**Review by aitchiss — APPROVED** [review]
Works as described 👍

**Review by msarit — APPROVED** [review]
Works as expected!

I noticed that, after creating a post as a non-admin member (with `limit_post_creation_to_admins` disabled), when the flag is re-enbled, I am unable to create new posts BUT able to edit my former posts. This is intended, right?

**Comment by jeremyf** [comment]
@msarit Let me rephrase what you said to make sure we don't have a "leak" in permissions:

With the `limit_post_creation_to_admins` feature flag enabled users can edit their existing posts but cannot create new ones.  Was that what you observed?

**Comment by msarit** [comment]
> @msarit Let me rephrase what you said to make sure we don't have a "leak" in permissions:
> 
> With the `limit_post_creation_to_admins` feature flag enabled users can edit their existing posts but cannot create new ones. Was that what you observed?

Yes, correct.

**Comment by jeremyf** [comment]
That is behavior I expected, but I'm confirming with @amywtlin that this is the desired behavior.  Note, the behavior is unrelated to this PR.

Thank you @msarit for bringing this up, always good to confirm permissions with folks.

**Comment by amywtlin** [comment]
> I noticed that, after creating a post as a non-admin member (with limit_post_creation_to_admins disabled), when the flag is re-enbled, I am unable to create new posts BUT able to edit my former posts. This is intended, right?

Yes - as @jeremyf mentioned, this would be intended behavior. The guiding principle Jeremy outlined in a Slack thread also aligns with how I think: “allow folks to continue to own that which they created.”
